### ISSUE TITLE
Forgetful browsing deletes storage in normal profile if browser launched in an Incognito profile via command line

### ISSUE THREAD

**Issue by whosithat** [brave/brave-browser#39107]
### Description

Hey everybody,

to face this issue, Brave Shields needs to be enabled and especially the option "forget me when I leave the page" needs to be enabled. 
Using then incognito mode, by starting it directly from the taskbar, it deletes all cookies in the default userprofile.

### Steps to reproduce

1. Open Brave with the default user profile (%localappdata%\BraveSoftware\Brave-Browser\User Data\Default)
2. Open any page and log into the site (e.g. facebook)
3. Close Brave
4. Open Brave Incognito by doing a right click into the taskbar brave icon
5. Close the Brave Incognito Windows
6. Open Brave again with the default user profile, the cookies should be deleted, you should recognize it by being logged out

### Actual result

The cookies in the default user profile gets deleted after closing the incognito mode windows.

### Expected result

The cookies of the default user profile stays untouched

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

Brave | 1.67.116 Chromium: 126.0.6478.71 (Offizieller Build) (64-Bit)
Revision | f4237fe448716e50cca6e827ff87ffd701c7d0c7
OS | Windows 10 Version 22H2 (Build 19045.4529)

### Channel information

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [X] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by ShivanKaul** [comment]
You're saying:

> to face this issue, Brave Shields needs to be enabled and especially the option "forget me when I leave the page" needs to be enabled.

If you've enabled this option, it is expected that closing the tab will delete all cookies. This has nothing to do with Incognito or Private Window, it's what "forget me when I leave the page" does.

**Comment by ShivanKaul** [comment]
> Open Brave Incognito by doing a right click into the taskbar brave icon
Close the Brave Incognito Windows
Open Brave again with the default user profile, the cookies should be deleted, you should recognize it by being logged out

When you open Brave again at this point, do you see your old tabs? Or does it start afresh because you launched a Private Window in between?

**Comment by whosithat** [comment]
> > Open Brave Incognito by doing a right click into the taskbar brave icon
> > Close the Brave Incognito Windows
> > Open Brave again with the default user profile, the cookies should be deleted, you should recognize it by being logged out
> 
> When you open Brave again at this point, do you see your old tabs? Or does it start afresh because you launched a Private Window in between?

Exactly, I can see all the old tabs and the session within the default user profile will be loaded completly successful again. 

Launching a private windows and closing it afterwards will delete all the site data of the open tabs within the default user profile.

Brave shield and the option "_forget me when I leave the page_" believes that the tabs have been closed, as the private window will be started without any tabs, so it thinks the default user session tabs have been closed, but they haven't. 

As far as I can result this from the user point of view, I believe that the problem is the architecture of the Brave shield, as they are running within the chrome backend and so it's not depending to the specific user profile, which is currently active. But that's just guessing.

**Comment by ShivanKaul** [comment]
Just a quick update, I want to repro this on my Windows machine which has been finicky, so will get back to this. I can't repro on MacOS.

**Comment by whosithat** [comment]
Hey @ShivanKaul,

any updates on this? I saw some progress in an linked item, but looks like that it's stuck now for a while..

**Comment by ShivanKaul** [comment]
It fell off the radar, we're working on an important anti-fingerprinting fix but should get to this soon.

**Comment by whosithat** [comment]
Hey @ShivanKaul,

any update on this? It looks like, that it's stuck again, at least the development item is waiting for an approval.

**Comment by whosithat** [comment]
Hi guys,

sorry, I really don't want to upset you, but is there any update on this?

**Comment by bsclifton** [comment]
Nice work @goodov! 😄 

@whosithat the fix is now merged meaning the next public Nightly build we do will have the fix (check back tomorrow). If no other action is taken, the fix will make it to release channel on ~March 4th 2025. 

We do have an opportunity to expedite that (ex: get fix in 1.75 / Feb 4) once we have confidence in the fix. Would be good to have it in Nightly for a few days to see how things sits before we make a call there

**Comment by MadhaviSeelam** [comment]
Verification `In progress` using
```
Brave | 1.76.50 Chromium: 133.0.6943.39 (Official Build) beta (64-bit)
-- | --
Revision | 5f3f8916ad419276d0ec62086e28074bf128a872
OS | Windows 11 Version 24H2 (Build 26100.2894)
```

Reproduced the issue in 1.67.116 Chromium: 126.0.6478.71 (Official Build) (64-bit) 

https://github.com/user-attachments/assets/3ef637cf-947c-4ed8-87a0-489bafbe3ba0
